src/tests/test_raft_repl_dev.cpp

```diff
@@ -331,45 +331,132 @@ TEST_F(RaftReplDevTest, All_Append_Fetch_Remote_Data) {
 
     g_helper->sync_for_verify_start();
 
-    // TODO: seems with filip and fetch remote, the data size is not correct;
     LOGINFO("Validate all data written so far by reading them");
     this->validate_all_data();
+
     g_helper->sync_for_cleanup_start();
 }
 
-TEST_F(RaftReplDevTest, All_ReplService) {
+// do some io before restart;
+TEST_F(RaftReplDevTest, All_restart_one_follower_inc_resync) {
     LOGINFO("Homestore replica={} setup completed", g_helper->replica_num());
     g_helper->sync_for_test_start();
-    auto repl_dev = dynamic_pointer_cast< RaftReplDev >(pick_one_db().repl_dev());
-    auto group_id = repl_dev->group_id();
-    auto my_id_str = repl_dev->my_replica_id_str();
 
-    auto leader = repl_dev->get_leader_id();
-    ASSERT_TRUE(leader != replica_id_t())
-        << "Error getting leader id for group_id=" << boost::uuids::to_string(group_id).c_str();
-    auto leader_str = boost::uuids::to_string(leader);
-    LOGINFO("Got raft leader {} for group {}", leader_str, group_id);
+    // step-0: do some IO before restart one member;
+    uint64_t exp_entries = 20;
+    if (g_helper->replica_num() == 0) {
+        g_helper->runner().set_num_tasks(20);
```

Comment:
The data channel and log channel use the same grpc connection actually and if there is any network issue, a new leader should be elected. For the network flip, I also discussed this with Sanel the other day, that we probably should do retry, this will make the logic complicated but if this network flip (network failure, but still leader doesn't change) is a real thing, we don't have choice but to handle it. I will mark it as a TODO also (to avoid comlicating things before we are 100% sure needed).